### ISSUE TITLE
Support for feature branches

### ISSUE THREAD

**Issue by aesopwolf** [zeit/release#29]
My team uses feature branches, and we also use the "Squash and merge" feature within GitHub. This makes our master branch look like this (by default):

![image](https://cloud.githubusercontent.com/assets/6847633/21669170/26041c22-d2be-11e6-9928-bbcf38902ed1.png)

So, do you think it would be possible for `release` to detect commit messages in that format (instead of using parenthesis such as `Error logging works now (patch)`)? The [Angular guidelines](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#type) could be used:

listed under `### Minor Changes`:

- feat
- feature

listed under `### Patches`:

- fix
- docs
- style
- refactor
- perf
- test
- chore

So anything the begins with those words would automatically be sorted to `minor` or `patch`.

**Comment by Tapppi** [comment]
Hi dropping in from the `@conventional-changelog` team. We have many tools for parsing git commits, doing releases etc. with different formats. I personally am most familiar with the angular convention, but I'm pretty sure you can find useful stuff from us. E.g. Our core changelog generator that is used by both `semantic-release` and `standard-version`.  Just gimme a holla if you want to know more.

**Comment by aesopwolf** [comment]
@Tapppi what do you guys use to detect breaking features? Or when you need to bump to the next MAJOR version number?

The most basic/naive solutions I can think of:

1. Include "(major)" in the commit message. This is what `release` is currently doing
2. Have the interactive CLI ask "Are there breaking changes in this release?"

I'm just wondering if there isn't a better/more-automated way though.

**Comment by Tapppi** [comment]
@aesopwolf The angular convention dictates that breaking change commits should have a footer of `BREAKING CHANGE:` and the rest of the commit message is reserved for that change.

See [the standard-version changelog for an example](https://github.com/conventional-changelog/standard-version/blob/master/CHANGELOG.md). So for example, the 4.0.0 has a bug fix
> include merge commits in the changelog

and if you go to the [commit linked to that change](https://github.com/conventional-changelog/standard-version/commit/b6e1562) you'll find the `BREAKING CHANGE` that is included as the 4.0.0 breaking change
> merge commits are now included in the CHANGELOG.


Docs on it can be found [in the standard-version commit guidelines](https://github.com/bcoe/conventional-changelog-standard/blob/master/convention.md#footer) or in the more detailed angular docs linked there.

**Comment by tleunen** [comment]
Indeed, it would be great if we could setup `release` to use the conventional-changelog format.
